addons/l10n_account_withholding/wizards/account_payment_register.py

```diff
@@ -0,0 +1,249 @@
+        """ We propose a default account by getting one from the latest payment which:
+            - Has the same payment method line id (and thus indirectly the same journal, and thus the same company)
+            - That payment method has no payment_account_id
+            - Yet the payment has an outstanding_account_id
+         """
+        for wizard in self:
+            latest_payment = self.env['account.payment'].search_read(
+                domain=[
+                    ('payment_method_line_id', '=', wizard.payment_method_line_id.id),
+                    ('payment_method_line_id.payment_account_id', '=', False),
+                    ('outstanding_account_id', '!=', False),
+                ],
+                fields=['outstanding_account_id'],
+                limit=1,
+                order='id desc'
+            )
+            if wizard.l10n_account_withholding_payment_account_id or not latest_payment:
+                wizard.l10n_account_withholding_outstanding_account = False  # we'll use the payment method one.
+            else:
+                wizard.l10n_account_withholding_outstanding_account = latest_payment[0]['outstanding_account_id'][0]
+
+    @api.depends('company_id')
+    def _compute_l10n_account_withholding_display_withholding(self):
+        for wizard in self:
+            available_withholding_taxes = self.env['account.tax'].search(wizard._get_withholding_tax_domain())
+            self.l10n_account_withholding_display_withholding = bool(available_withholding_taxes)
+
+    @api.depends('line_ids')
+    def _compute_from_lines(self):
+        """
+        Extended in order to pre-populate the withholding lines based on the taxes set on the products of the invoice.
+
+        Products can have withholding taxes assigned to them.
+        These will not appear on the invoice, but are intended to be used here to pre-populate the withholding tax lines.
+        We will make one line per withholding tax, and pre-set the base amount as the sum of the lines with a product having this tax set.
+        """
+        # EXTEND account
+        super()._compute_from_lines()
+        for wizard in self:
+            if wizard.country_code == 'AR':
+                wizard.l10n_account_withholding_line_ids = False
+                continue
+
+            withholding_line_creation_vals = []
+            if wizard.can_edit_wizard:
+                withholding_line_amounts = defaultdict(int)
+                batch_result = wizard.batches[0]
+                for move in batch_result['lines'].move_id:
+                    for line in move.invoice_line_ids:
+                        taxes = line.product_id.taxes_id if move.is_sale_document(include_receipts=True) else line.product_id.supplier_taxes_id
+                        withholding_taxes = taxes.filtered_domain(wizard._get_withholding_tax_domain(move.company_id))
+                        for withholding_tax in withholding_taxes:
```

Comment:
This is the fourth loop, I guess the ORM will struggle a bit. Can we batch something?

Reply:
In practice, you won't have multiple wizard and this also doesn't work in batch mode.
I may be wrong but these loops shouldn't run too much and what's done here isn't super heavy :thinking: 
If you have any idea on how to batch it would be welcome though :)

The few things we could batch should already be handled by the orm caching.